### PR TITLE
Disable showing offers based on client config ui predicates

### PR DESCRIPTION
**Pull request by adolgachev**

### CODE REVIEW

File: src/runtime/offers-flow.js [hunk 1]
```diff
@@ -225,19 +231,22 @@ export class OffersFlow {
     return Promise.resolve();
   }
 
+  shouldShow_(clientConfig) {
```
Comment (mmmicedcoffee) [inline src/runtime/offers-flow.js:234]: is it worth naming this method "shouldShowAutoPrompt" for additional clarity?

Comment (mmmicedcoffee) [reply src/runtime/offers-flow.js:234]: also, please add a JSDoc comment so there's type information for the argument.

File: src/runtime/offers-flow.js [hunk 2]
```diff
@@ -225,19 +231,22 @@ export class OffersFlow {
...
+      clientConfig.uiPredicates.canDisplayAutoPrompt
+    );
+  }
+
   /**
    * Gets the URL that should be used for the activity iFrame view.
-   * @param {!Promise<../model/client-config.ClientConfig>} clientConfigPromise
-   * @return {!Promise<string>}
+   * @param {!../model/client-config.ClientConfig} clientConfig
+   * @return {!string}
```
Comment (mmmicedcoffee) [inline src/runtime/offers-flow.js:244]: primitive types are non-nullable by default, so this type annotation can be simplified to "@return {string}"

File: src/runtime/offers-flow.js [hunk 3]
```diff
@@ -121,17 +121,19 @@ export class OffersFlow {
     this.skus_ = feArgsObj['skus'] || [ALL_SKUS];
 
     /** @private @const {!Promise<!ActivityIframeView>} */
```
Comment (mmmicedcoffee) [inline src/runtime/offers-flow.js:123]: the type needs to be updated to reflect that the Promise can return null (!Promise<?ActivityIframeView>)

File: src/runtime/offers-flow.js (hunk 1 above)
Comment (adolgachev) [reply src/runtime/offers-flow.js:234]: Actually intentionally keeping it more generic (just tied to showing of offers in this context)  as just re-using the UI predicates which both return the same thing but eventually may be tweaked and expanded.

Comment (mmmicedcoffee) [reply src/runtime/offers-flow.js:241]: sounds good!

File: src/runtime/offers-flow.js [hunk 4]
```diff
@@ -120,18 +120,20 @@ export class OffersFlow {
...
-    ).then((url) => {
-      return new ActivityIframeView(
-        this.win_,
-        this.activityPorts_,
-        feUrl(url),
-        feArgsObj,
-        /* shouldFadeBody */ true
-      );
-    });
+    /** @private @const {!Promise<!ActivityIframeView|null>} */
```
Comment (mmmicedcoffee) [inline src/runtime/offers-flow.js:123]: is "!ActivityIframeView|null" different from "?ActivityIframeView"? I thought the "?" indicates "|null".

**Review by mmmicedcoffee — APPROVED** [review]

File: src/runtime/offers-flow.js (hunk 4 above)
Comment (adolgachev) [reply src/runtime/offers-flow.js:123]: I was intending for explicit null (v.s. undefined like typescript) but, it doesn't seem to make a difference here. The docs are more explicit for behavior for params but I'll move to the simpler syntax in any case.

**Comment by fikus** [comment]
Issue reference: b/190849246

**Pull request merged**